app/src/main/java/com/example/egida/domain/useCase/day/DayRepository.kt

```diff
@@ -7,5 +7,5 @@ interface DayRepository {
     suspend fun getDay()
     var day: Flow<Day>
     fun createDay(day: Flow<Day>)
-//    I'm pretty sure, that there is no real need to pass flow inside of repository.
+    fun updateValueDay(day: Flow<Day>)
```

Comment:
You are sharing day flow from one repository, to another one. Terrible approach. You should think about data flow. It's better to write a diagram. I suppose, you should just pass Day as a param here. And your ViewModel should not just 'shareIn' day variable, but collect and have local instanse. At least, because you are modifying it here. 

Better approach - update value in repository itself and return correct answer apward.